internal/cmd/feedback/command.go

```diff
@@ -10,71 +10,77 @@ import (
 
 	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
-	v1 "github.com/confluentinc/cli/internal/pkg/config/v1"
+	"github.com/confluentinc/cli/internal/pkg/errors"
 	"github.com/confluentinc/cli/internal/pkg/form"
-	"github.com/confluentinc/cli/internal/pkg/log"
+	"github.com/confluentinc/cli/internal/pkg/output"
 	pversion "github.com/confluentinc/cli/internal/pkg/version"
 )
 
-func NewFeedbackCmd(cfg *v1.Config, prerunner pcmd.PreRunner) *cobra.Command {
-	prompt := form.NewPrompt(os.Stdin)
-	return NewFeedbackCmdWithPrompt(cfg, prerunner, prompt)
+type command struct {
+	*pcmd.AuthenticatedCLICommand
+	form.Prompt
 }
 
-func NewFeedbackCmdWithPrompt(cfg *v1.Config, prerunner pcmd.PreRunner, prompt form.Prompt) *cobra.Command {
-	cmd := pcmd.NewAnonymousCLICommand(
-		&cobra.Command{
-			Use:   "feedback",
-			Short: "Submit feedback about the " + pversion.FullCLIName + ".",
-			RunE: func(cmd *cobra.Command, _ []string) error {
-				msg, err := getFeedback(prompt)
-				if err != nil {
-					return err
-				}
-				if len(msg) > 0 {
-					if err := sendFeedback(cfg, cmd, msg); err != nil {
-						return err
-					}
-					fmt.Println("Thanks for your feedback.")
-				}
-				return nil
-			},
-			Args: cobra.NoArgs,
-		},
-		prerunner)
+const (
+	maxLength = 2000
+)
+
+func New(prerunner pcmd.PreRunner) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:         "feedback",
+		Short:       fmt.Sprintf("Submit feedback about the %s.", pversion.FullCLIName),
+		Args:        cobra.NoArgs,
+		Annotations: map[string]string{pcmd.RunRequirement: pcmd.RequireCloudLogin},
+	}
+
+	c := &command{
+		AuthenticatedCLICommand: pcmd.NewAuthenticatedCLICommand(cmd, prerunner),
+		Prompt:                  form.NewPrompt(os.Stdin),
+	}
+	cmd.RunE = c.feedback
 
-	return cmd.Command
+	return cmd
 }
 
-func getFeedback(prompt form.Prompt) (string, error) {
-	fmt.Println("Please confirm that your feedback will not contain any " +
-		"sensitive information such as API-keys, unencrypted tokens, etc.")
-	fmt.Print("Type \"y\" to confirm or anything else to exit: ")
-	ack, err := prompt.ReadLine()
+func (c *command) feedback(_ *cobra.Command, _ []string) error {
+	msg, err := getFeedback(c.Prompt)
 	if err != nil {
-		return "", err
+		return err
 	}
-	if ack != "y" {
+	if len(msg) > 0 {
+		if err := c.sendFeedback(msg); err != nil {
+			return err
+		}
+		fmt.Println("Thanks for your feedback.")
+	}
+	return nil
+}
+
+func getFeedback(prompt form.Prompt) (string, error) {
+	f := form.New(form.Field{
+		ID: "proceed",
+		Prompt: "Please confirm that your feedback will not contain any " +
+			"sensitive information such as API-keys, unencrypted tokens, etc.",
+		IsYesOrNo: true,
+	})
+	if err := f.Prompt(prompt); err != nil || !f.Responses["proceed"].(bool) {
+		output.Println("Not proceeding with feedback")
 		return "", err
 	}
-	fmt.Println("Enter feedback: ")
+	output.Println("Enter feedback (maximum length of 2,000 characters): ")
```

Comment:
99% of the time, users won't exceed this limit. Let's make sure the backend returns a nice error if this happens though. (Also, if we change this number in the future, we don't want older versions of the CLI to still say 2000)
```suggestion
	output.Print("Enter feedback: ")
```